sound/soc/sof/ipc4-topology.c

```diff
@@ -1600,16 +1601,19 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 	void *cfg = process->ipc_config_data;
 	const struct snd_kcontrol_new *kc;
 	struct soc_bytes_ext *sbe;
+	int data_size;
 	void *data;
 	int ret, i;
 
 	available_fmt->ref_audio_fmt = &available_fmt->base_config->audio_fmt;
 
 	/*
-	 * Output format is optional for process modules.
-	 * Process modules setup the output format based on audio format tokens in topology.
+	 * Output format is only valid and optional for modules with base module config +
```

Comment:
i'd switch around the optional and only valid bits. "Output format is optional and only valid ...." and the check below as well